cmd/helm/root.go

```diff
@@ -92,7 +93,7 @@ func newRootCmd(actionConfig *action.Configuration, out io.Writer, args []string
 			completion.CompDebugln(fmt.Sprintf("About to call kube client for namespaces with timeout of: %d", to))
 
 			nsNames := []string{}
-			if namespaces, err := client.CoreV1().Namespaces().List(metav1.ListOptions{TimeoutSeconds: &to}); err == nil {
+			if namespaces, err := client.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{TimeoutSeconds: &to}); err == nil {
```

Comment:
Should this use `context.WithTimeout`?

```suggestion
			ctx, cancel := context.WithTimeout(context.Background(), time.Second * time.Duration(to))
			defer cancel()
			if namespaces, err := client.CoreV1().Namespaces().List(ctx, metav1.ListOptions{TimeoutSeconds: &to}); err == nil {
```

And for other places that `context.Background()` is used, is it truly the intention for those functions to be allowed to wait forever? Or should we use `context.TODO()` as a reminder that we plan to add cancellation?

Reply:
to retain backwards compatibility, yes.

Reply:
We use semver which is why all these workarounds are necessary.